### PR TITLE
Add events to timing diagram

### LINKED ISSUE #562: Add events to new timing diagram
Within the Cromwell metadata, there are `executionEvents` within tasks, which could provide more information to the user if included within the timing diagram.

### PR DESCRIPTION
**Pull request by rsasch**
Adds granularity to timing diagram for tasks/subworkflows in Cromwell.

Closes #562

### CODE REVIEW

File: ui/src/app/job-details/tabs/timing-diagram/timing-diagram.component.ts [hunk 1]
```diff
@@ -20,23 +21,42 @@ export class JobTimingDiagramComponent implements OnInit {
...
+
     this.timelineChart.options = {
-      title: 'Tasks',
-      width: 800,
-      height: counter * 50
+      timeline: {
+        showRowLabels: false,
+        showBarLabels: false
+      },
+      width: 1470,
```
Comment (calbach) [inline ui/src/app/job-details/tabs/timing-diagram/timing-diagram.component.ts]: Where did this value come from? Please add a comment, e.g. if it's matching something in CSS somewhere.

Comment (rsasch) [reply ui/src/app/job-details/tabs/timing-diagram/timing-diagram.component.ts]: I've added in some logic to find the browser width and calculate the graph width based on that.

File: ui/src/app/job-details/tabs/timing-diagram/timing-diagram.component.ts [hunk 2]
```diff
@@ -20,23 +21,42 @@ export class JobTimingDiagramComponent implements OnInit {
...
+
     this.timelineChart.options = {
-      title: 'Tasks',
-      width: 800,
-      height: counter * 50
+      timeline: {
+        showRowLabels: false,
+        showBarLabels: false
+      },
+      width: this.tabWidth - 65,
```
Comment (calbach) [inline ui/src/app/job-details/tabs/timing-diagram/timing-diagram.component.ts]: Looks like this only runs on `ngOnInit` so I'm not sure I'd expect this to resize itself properly. Perhaps a percentage based width would work better (if supported?).

Anyways - I can't really test this out, so I'll leave it to you or another reviewer to ensure this is functional

File: ui/src/app/job-details/tabs/timing-diagram/timing-diagram.component.ts [hunk 3]
```diff
@@ -20,23 +21,42 @@ export class JobTimingDiagramComponent implements OnInit {
...
+      height: (counter * 42) + 100,
```
Comment (calbach) [inline ui/src/app/job-details/tabs/timing-diagram/timing-diagram.component.ts]: nit: "counter" is pretty generic - I'd name it something more semantically relevant, such as `numRows`, or whatever it's actually referring to.

File: ui/src/app/job-details/tabs/timing-diagram/timing-diagram.component.ts (hunk 2 above)
Comment (rsasch) [reply ui/src/app/job-details/tabs/timing-diagram/timing-diagram.component.ts]: It isn't set to re-size, just to figure out how wide to make it when the component is first rendered.

File: ui/src/app/job-details/tabs/timing-diagram/timing-diagram.component.ts (hunk 3 above)
Comment (rsasch) [reply ui/src/app/job-details/tabs/timing-diagram/timing-diagram.component.ts]: good point – changed

File: ui/src/app/job-details/job-details.component.ts [hunk 4]
```diff
@@ -26,8 +26,8 @@ export class JobDetailsComponent implements OnInit {
   }
 
   hasTabs(): boolean {
-    if (this.job.inputs || this.job.outputs) {
-      return true;
+    if (( this.job.inputs && Object.keys(this.job.inputs).length !== 0) || (this.job.outputs  && Object.keys(this.job.outputs).length !== 0)) {
```
Comment (cjllanwarne) [inline ui/src/app/job-details/job-details.component.ts]: Is this change related?

File: ui/src/app/job-details/tabs/tabs.component.ts [hunk 5]
```diff
@@ -54,6 +55,9 @@ export class JobTabsComponent implements OnInit, OnChanges {
     if (this.hasCallCachedTask() || this.hasScatteredTask()) {
       this.displayedColumns.splice(1, 0, "taskInfoIcons");
     }
+    if (this.tabsPanel) {
+      this.tabWidth = this.tabsPanel._viewContainerRef.element.nativeElement.clientWidth;
```
Comment (rexwangcc) [inline ui/src/app/job-details/tabs/tabs.component.ts]: Does this add the responsiveness to the timing panel?

Comment (rsasch) [reply ui/src/app/job-details/tabs/tabs.component.ts]: It allows the timing diagram component to determine the width to be initially rendered at.  It does not change if the size of the browser window is changed without a reload.

File: ui/src/app/job-details/job-details.component.ts (hunk 4 above)
Comment (rsasch) [reply ui/src/app/job-details/job-details.component.ts]: tangentially – this function will determine if the tab group, which the timing diagram is in, will be displayed

File: ui/src/app/job-details/tabs/tabs.component.ts (hunk 5 above)
Comment (rsasch) [reply ui/src/app/job-details/tabs/tabs.component.ts]: no, it's just passed along to the function that builds the timing diagram when the component is loaded
